4_naming_conventions/43_file_names.md

```diff
@@ -58,3 +59,154 @@ regular expression:
+                                        'LoongArch64' | 'Ebc'   
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+Example:
+   PCD/Dxe/Pcd.inf
+```
+
+#### 4.4.5.3 EDK II INF file within a Library instance
+```
+<Phase><CpuArch><Vendor><LibraryClassName><Dependency>.inf
+   <Phase>              REQUIRED     Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                                     StandaloneMm, Smm, Uefi.
+   <CpuArch>            OPTIONAL     The <CpuArch> is represented with a BNF,
+                                     <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                               'LoongArch64' | 'Ebc'
+                                     <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                                     Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>             OPTIONAL     *
+   <LibraryClassName>   REQUIRED     *
+   <Dependency>         OPTIONAL     * (Typically name of PPI, Protocol, LibraryClass
+                                       that the implementation is layered on top of)
+
+Example:
+   - SmmAmdSmmCpuFeaturesLib.inf
+   - SmmIntelSmmCpuFeaturesLib.inf
+   - SmmIa32_X64SmmCpuFeaturesLib.inf
+```
+
+#### 4.3.5.4 EDK II source files within a Library/Module instance
+
+In generally, the file name is constructed as below:
+```
+
+<CpuArch><Vendor><FileName>.c
+<CpuArch><Vendor><FileName>.h
+
+   <CpuArch>   OPTIONAL   The <CpuArch> is represented with a BNF,
+                          <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                    'LoongArch64' | 'Ebc'
+                          <CpuArch> ::= <arch>['_'<arch>]*
+
+                          Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>    OPTIONAL   *
+   <FileName>  REQUIRED   *
+
+Example:
+   SmmCpuFeatureLib.c
+   SmmCpuFeatureLibCommon.c
+   IntelSmmCpuFeaturesLib.c
+   AmdSmmCpuFeaturesLib.c
+   Ia32_X64SmmCpuFeaturesLib.c
+
+```
+
+While there is a processor arch or vendor supported in the module:
+```
+<CpuArch>/<FileName>.c
+<CpuArch>/<FileName>.h
+<CpuArch>/<FileName>.nasm
+<CpuArch>/<FileName>.S
+<CpuArch>/<Vendor>/<FileName>.c
+<CpuArch>/<Vendor>/<FileName>.h
+<CpuArch>/<Vendor>/<FileName>.nasm
+<CpuArch>/<Vendor>/<FileName>.S
+
+   <CpuArch>   OPTIONAL   The <CpuArch> is represented with a BNF,
+                          <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                    'LoongArch64' | 'Ebc'
+                          <CpuArch> ::= <arch>['_'<arch>]*
+
+                          Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>    OPTIONAL   *
+   <FileName>  REQUIRED   *
+
+Example:
+   IA32/SmiException.nasm
+   X64/SmiException.nasm
+   IA32_X64/IntelSmmCpuFeaturesLib.c
+   IA32_X64/AmdSmmCpuFeaturesLib.c

```

Comment:
Did you modify the directory path from IA32_X64 to Ia32_X64 in the examples above? For some reason, I still see IA32.

Reply:
Just pushed!

Reply:
Fix at commit bcb249b1788d7b11b182e2c2e1043fc7793a071a